libvast/test/view.cpp

```diff
@@ -45,6 +45,46 @@ TEST(container view) {
   CHECK_EQUAL(i - j, xs.size() - 1);
 }
 
+TEST(set view) {
+  auto xs = set{42, true};
+  auto v = make_view(xs);
+  REQUIRE_EQUAL(v->size(), xs.size());
+  auto i = v->begin();
+  // FIXME: we currently use a std::set instead of a steady set, which relies
+  // on a non-deterministic ordering. This unit test may fail on some platforms
+  // until we have switched to steady_set
+  CHECK_EQUAL(*i, v->at(0));
+  CHECK_EQUAL(*i, make_data_view(42));
+  ++i;
+  CHECK_EQUAL(*i, v->at(1));
+  CHECK_EQUAL(*i, make_data_view(true));
+  ++i;
+  CHECK_EQUAL(i, v->end());
+}
+
+TEST(table view) {
+  auto xs = table{{42, true}, {84, false}};
+  auto v = make_view(xs);
+  REQUIRE_EQUAL(v->size(), xs.size());
+  auto i = v->begin();
+  // FIXME: we currently use a std::table instead of a steady table, which relies
+  // on a non-deterministic ordering. This unit test may fail on some platforms
+  // until we have switched to steady_map.
+  {
```

Comment:
The manual index and iterator increments make the code more verbose than it needs to be IMO. I would use an extra container for storing the expected values to streamline the test:

```cpp
MESSAGE("check values in the view");
std::vector<std::pair<int, bool>> anticipated{{42, true}, {84, false}};
for (size_t i = 0; i < anticipated.size(); ++i) {
  auto& [key, value] = v->at(i);
  auto& [anticipated_key, anticipated_value] = anticipated[i];
  CHECK_EQUAL(key, make_data_view(anticipated_key));
  CHECK_EQUAL(value, make_data_view(anticipated_value));
}
MESSAGE("check iterator semantics of the view");
CHECK_EQUAL(std::next(v->begin(), 2), v->end());
```